observability/test/run_o11y_tests.py

```diff
@@ -0,0 +1,1174 @@
+            yaml_file_data = yaml_file_data.replace(orig, replacement)
+        logger.info('Applying deployment from yaml...')
+        logger.info(yaml_file_data)
+        tmp_file_path = '/tmp/%s:%s:%s.yaml' % (self.args.server_lang,
+                                                self.args.client_lang,
+                                                self.args.gke_resource_identifier)
+        with open(tmp_file_path, 'w') as f:
+            f.write(yaml_file_data)
+        # Deploy to GKE cluster
+        kubernetes_utils.create_from_yaml(self.k8s_client, tmp_file_path, verbose=True)
+        logger.info('Waiting for pods to be ready...')
+        timeout_target = int(time.time()) + WAIT_SECS_GKE_DEPLOYMENT
+        while True:
+            try:
+                if int(time.time()) > timeout_target:
+                    break
+                time.sleep(10)
+                self.get_active_pod_data()
+                return
+            except Exception:
+                logger.error(traceback.format_exc())
+        raise Exception('GKE resources still not ready after %d seconds' % WAIT_SECS_GKE_DEPLOYMENT)
+
+    def clean_up_gke_resources(self) -> None:
+        f = os.popen(os.path.join(os.path.dirname(__file__), 'cleanup.sh ' +
+                                  self.args.gke_resource_identifier))
+        output = f.read()
+        logger.info(output)
+
+    def get_active_pod_data(self) -> None:
+        ret = self.k8s_core_v1.list_namespaced_pod(self.SERVER_NAMESPACE)
+        for i in ret.items:
+            logger.debug('%s\t%s\t%s' % (i.status.pod_ip, i.metadata.name, i.status.phase))
+            if self.server_pod_name and i.metadata.name != self.server_pod_name:
+                raise Exception('Somehow found more than 1 pod in the server namespace.')
+            if i.status.phase != 'Running':
+                raise Exception('%s is not ready yet.' % i.metadata.name)
+            self.server_pod_name = i.metadata.name
+            self.server_pod_ip = i.status.pod_ip
+        ret = self.k8s_core_v1.list_namespaced_pod(self.CLIENT_NAMESPACE)
+        for i in ret.items:
+            logger.debug('%s\t%s\t%s' % (i.status.pod_ip, i.metadata.name, i.status.phase))
+            if self.client_pod_name and i.metadata.name != self.client_pod_name:
+                raise Exception('Somehow found more than 1 pod in the client namespace.')
+            if i.status.phase != 'Running':
+                raise Exception('%s is not ready yet.' % i.metadata.name)
+            self.client_pod_name = i.metadata.name
+        if not self.server_pod_name:
+            raise Exception('No server pod found in %s.' % self.SERVER_NAMESPACE)
+        if not self.client_pod_name:
+            raise Exception('No client pod found in %s.' % self.CLIENT_NAMESPACE)
+
+    def _test_pid(self, s):
+        if m := re.match('PID=(\d+)', s):
+            return m.group(1)
+        return False
+
+    def exec_commands_in_pod(self,
```

Comment:
Why did you go with this approach? I'd much rather we pass args to the container when it starts.

Reply:
Oh, I didn't notice you weren't using docker locally when running locally. It should really be docker there, and we get rid of the `sleep inf` goo.

Reply:
This has been implemented.

Now individual language needs to supply a `Dockerfile` that has a `run.sh` `ENTRYPOINT` which takes a pre-defined set of parameters to determine how the interop client/server is executed.

This allows this main test driver to just execute `docker run <docker image> <additional parameters>` and don't need to worry about knowing individual language's implementation.